torch_geometric/explainability/algo/captumexplainer.py

```diff
@@ -0,0 +1,147 @@
+import torch
+
+from torch_geometric.explainability.algo.base import ExplainerAlgorithm
+from torch_geometric.explainability.explanations import Explanation
+from torch_geometric.explainability.utils import CaptumModel
+
+
+class CaptumExplainer(ExplainerAlgorithm):
+    """Wrapper for captum explainers.
+
+    Wraper for captum explainers.
+
+    Note:
+        This algorithm expect the model to be callable with the following
+        signature: model(x, edge_index, **kwargs).
+
+
+    Args:
+        method (str, optional): name of the captum explainer.
+        Defaults to "Saliency".
+    """
+    def __init__(self, method: str = "Saliency") -> None:
+
+        super().__init__()
+        self.method = method
+
+    def explain(
+        self,
+        x: torch.Tensor,
+        edge_index: torch.Tensor,
+        target: torch.Tensor,
+        model: CaptumModel,
+        mask_type: str = "node",
+        **kwargs,
+    ) -> Explanation:
+
+        from captum import attr  # noqa
+
+        # non optimal for now, but captum requires the model to instantiate
+        # the explanation algorithm
+        explainer = getattr(attr, self.method)(model)
+        input, additional_forward_args = self._arange_inputs(
+            x, edge_index, mask_type, **kwargs)
+
+        attributions = explainer.attribute(
+            **self._arange_args(input, target, additional_forward_args))
+
+        return self._create_explanation_from_masks(x, edge_index, attributions)
+
+    def supports(
```

Comment:
wonder if we should add some test coverage. I'm also not too sure. But perhaps use some mocks?

Reply:
I would first check the shape of the explanations and the range of the values. Maybe we could try to implement tests based on `BA2-shapes` (the house vs cycle task)?